File: sach/de3361.cpp
//DE3361.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
int n,q;
int x[100005];
int main() {
    cin>>n>>q;
    int i,j,a,b;
    for (i=1; i<=n; i++) cin>>x[i];
    sort(x+1,x+n+1);
    for (int t=1; t<=q; t++) {
        cin>>a>>b;
        i=lower_bound(x+1,x+n+1,a)-x;
        j=upper_bound(x+1,x+n+1,b)-x;
        cout<<j-i<<endl;
    }
    return 0;
}

File: sach/de3361.py
//DE3361.PY - Tran Huu Nam - hsgtin.vn - 2024
import bisect
n, q = map(int, input().split())
x = [0] * 100005
for i in range(1, n + 1):
    x[i] = int(input())
x = sorted(x[1:n+1])
for _ in range(q):
    a, b = map(int, input().split())
    i = bisect.bisect_left(x, a)
    j = bisect.bisect_right(x, b)
    print(j - i)

